src/main/java/com/epam/idea/core/service/impl/IdeaServiceImpl.java

```diff
@@ -81,6 +81,15 @@ public Idea update(final long ideaId, final Idea source) {
 	}
 
 	@Override
+	public List<Idea> findIdeasByTagId(final long tagId) {
+		List<Idea> ideas = ideaRepository.findByTagId(tagId);
+		ideas.forEach(idea -> {
+			Hibernate.initialize(idea.getRelatedTags());
```

Comment:
I think Hibernate will try to initialize `ideasWithTag` field inside every tag in `idea.getRelatedTags()` collection. But we only need tag `name`, so I think just call `idea.getRelatedTags()` will be enough.